src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -121,20 +116,15 @@ class DrawCall {
     this.offsetVertexBuffer = offsetVertexBuffer;
     this.keepInstanceStepModeBufferInRange = keepInstanceStepModeBufferInRange;
 
-    // Since vertexInIndexBuffer is mutable, index buffer generating should be deferred to right before calling draw
+    // Since vertexInIndexBuffer is mutable, generation of the index buffer should be deferred to right before calling draw
 
     // Generate vertex buffer
-    if (keepInstanceStepModeBufferInRange) {
-      // Surpress partialLastNumber for the first vertex buffer, aka the instance step mode buffer
-      this.vertexBuffers = [
-        this.generateVertexBuffer(vertexArrays[0], false),
-        ...vertexArrays
-          .slice(1, vertexArrays.length)
-          .map(v => this.generateVertexBuffer(v, partialLastNumber)),
-      ];
-    } else {
-      this.vertexBuffers = vertexArrays.map(v => this.generateVertexBuffer(v, partialLastNumber));
-    }
+    this.vertexBuffers = vertexArrays.map((v, i) =>
+      i === 0 && keepInstanceStepModeBufferInRange
```

Comment:
nit: using ternaries for control flow is a bit weird, can we use an if? You can add {} to make a real body for the lambda function.

Reply:
Done.